### PR TITLE
Feature - Creating of public profile routes, showing the user's submissions 

### PR DESCRIPTION
**Pull request by nspilman**
This diff adds public profiles, showing the covers the user has submitted.

### CODE REVIEW

File: queries/index.ts [hunk 1]
```diff
@@ -15,6 +15,8 @@ export enum Views {
   VotesDiffsWithAverage = "votes_diff_with_average",
   Submissions = "submissions_view",
   VoteBreakdownBySong = "vote_breakdown_by_song",
+  PublicSignups = "public_signups",
+  PublicSubmissions = "public_submissions",
```
Comment (BBtxt) [inline queries/index.ts:19]: So you added the PublicSignUps, and PublicSubmissions to this Enum for use in the [username] route. Why do you have to use an enum here?

File: pages/profile/[username].tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  return {
+    // Passed to the page component as props
+    props: {
+      username,
+      data,
+    },
+  };
+}
+
+export default function Profile(props: { username: string; data: any }) {
```
Comment (BBtxt) [inline pages/profile/[username].tsx:41]: So did you rip this function from the profiles component? I was i got really confused if we had to get rid of the router in that component.

File: pages/profile/[username].tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    paths: data?.map(({ username }) => ({
+      params: { username },
+    })),
+    fallback: false, // can also be true or 'blocking'
+  };
+  return payload;
+}
+
+// `getStaticPaths` requires using `getStaticProps`
+export async function getStaticProps({
```
Comment (BBtxt) [inline pages/profile/[username].tsx:22]: For the getStaticProps, Why are there two Params? passed into the function parameters.  These params is it coming from the getStaticPath? And it is one of the many Usernames from the view table?

**Review by BBtxt — COMMENTED** [review]
It's really nice to see how you did this. I feel like I kinda got the gist but blew the execution.

File: queries/index.ts (hunk 1 above)
Comment (nspilman) [reply queries/index.ts:19]: The idea is that I don't want to select strings, I want to select enum values that represent strings. Reason being, it's way more typesafe. 

I could also pull this off by properly configuring my Supabase instance, but...

File: pages/profile/[username].tsx (hunk 3 above)
Comment (nspilman) [reply pages/profile/[username].tsx:24]: There's only one param! the second thing you're seeing is the type declaration. The `username` is the only thing passed from `getStaticPaths` to `getStaticProps`

File: pages/profile/[username].tsx (hunk 2 above)
Comment (nspilman) [reply pages/profile/[username].tsx:43]: Lol actually yeah - the component name here doesn't matter at all though. All we need is something default exported.

**Pull request merged**